cmd/registry/cmd/sync/sync.go

```diff
@@ -0,0 +1,241 @@
+	s = strings.ReplaceAll(s, ".", "-")
+	return strings.ToLower(s)
+}
+
+func deployments(ctx context.Context, org string) ([]*apigee.GoogleCloudApigeeV1Deployment, error) {
+	apg, err := apigee.NewService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := apg.Organizations.Deployments.List(org).Context(ctx).Do()
+	return resp.Deployments, err
+}
+
+type apiProxyRevision struct {
+	Basepaths   []string `json:"basepaths,omitempty"`
+	ProxyName   string   `json:"name,omitempty"`
+	RevisionID  string   `json:"revision,omitempty"`
+	Description string   `json:"description,omitempty"`
+}
+
+func revision(ctx context.Context, revision, token string) (*apiProxyRevision, error) {
+	req := &http.Request{
+		Method: "GET",
+		URL: &url.URL{
+			Scheme: "https",
+			Host:   "apigee.googleapis.com",
+			Path:   fmt.Sprintf("/v1/%s", revision),
+		},
+		Header: http.Header{
+			"Authorization": []string{fmt.Sprintf("Bearer %s", token)},
+		},
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	rev := &apiProxyRevision{}
+	return rev, json.NewDecoder(resp.Body).Decode(rev)
+}
+
+func envgroups(ctx context.Context, org string) ([]*apigee.GoogleCloudApigeeV1EnvironmentGroup, error) {
+	apg, err := apigee.NewService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := apg.Organizations.Envgroups.List(org).Context(ctx).Do()
+	return resp.EnvironmentGroups, err
+}
+
+func attachments(ctx context.Context, group string) ([]*apigee.GoogleCloudApigeeV1EnvironmentGroupAttachment, error) {
+	apg, err := apigee.NewService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := apg.Organizations.Envgroups.Attachments.List(group).Context(ctx).Do()
+	return resp.EnvironmentGroupAttachments, err
+}
+
+type envMap struct {
+	hostnames map[string][]string
+	envgroup  map[string]string
+}
+
+func (m *envMap) Hostnames(env string) ([]string, bool) {
+	if m.hostnames == nil {
+		return nil, false
+	}
+
+	v, ok := m.hostnames[env]
+	return v, ok
+}
+
+func (m *envMap) Envgroup(hostname string) (string, bool) {
+	if m.envgroup == nil {
+		return "", false
+	}
+
+	v, ok := m.envgroup[hostname]
+	return v, ok
+}
+
+func newEnvMap(ctx context.Context, org string) (*envMap, error) {
+	groups, err := envgroups(ctx, org)
+	if err != nil {
+		return nil, err
+	}
+
+	m := &envMap{
+		hostnames: make(map[string][]string),
+		envgroup:  make(map[string]string),
+	}
+
+	for _, group := range groups {
+		envgroup := fmt.Sprintf("%s/envgroups/%s", org, group.Name)
+		attachments, err := attachments(ctx, envgroup)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, attachment := range attachments {
+			for _, hostname := range group.Hostnames {
+				m.hostnames[attachment.Environment] = append(m.hostnames[attachment.Environment], hostname)
+				m.envgroup[hostname] = envgroup
+			}
+		}
+	}
+
+	return m, nil
+}
+
+func bundle(ctx context.Context, revision, token string) ([]byte, error) {
```

Comment:
This isn't used anywhere, but I kept it just in case we decide to download proxy bundles for some reason.